Google.Api.Generator/Generation/MethodDetails.cs

```diff
@@ -115,20 +143,40 @@ public Lro(ServiceDetails svc, MethodDescriptor desc) : base(svc, desc)
                 OperationResponseTyp = ProtoTyp.Of(responseTypeMsg);
                 OperationMetadataTyp = ProtoTyp.Of(metadataTypeMsg);
                 SyncReturnTyp = Typ.Generic(typeof(Operation<,>), OperationResponseTyp, OperationMetadataTyp);
-                LroSettingsName = $"{desc.Name}OperationsSettings";
-                LroClientName = $"{desc.Name}OperationsClient";
-                SyncPollMethodName = $"PollOnce{SyncMethodName}";
-                AsyncPollMethodName = $"PollOnce{AsyncMethodName}";
             }
+
             public override Typ ApiCallTyp { get; }
             public override Typ SyncReturnTyp { get; }
-            public string LroSettingsName { get; }
-            public string LroClientName { get; }
-            public string SyncPollMethodName { get; }
-            public string AsyncPollMethodName { get; }
-            public Typ OperationTyp => SyncReturnTyp;
-            public Typ OperationResponseTyp { get; }
-            public Typ OperationMetadataTyp { get; }
+            public override Typ OperationResponseTyp { get; }
+            public override Typ OperationMetadataTyp { get; }
+        }
+
+        /// <summary>
+        /// Details about an LRO method declared in a Discovery-sourced proto
+        /// using an operation_service annotation. 
+        /// </summary>
+        public sealed class DiscoveryLro : Lro
```

Comment:
I'd rather we not call it DiscoveryLro, because the connection with a discovery document is gone by the time we encounter it. 
 Diregapic or ExtendedLro perhaps?

Reply:
Diregapic feels more obscure - and still implies Discovery, but also implying REST/JSON. I'm not keen on ExtendedLro either, as that sounds like it's more capable.

How about just NonStandardLro? That's really why we need it - because LROs weren't standardised when these APIs were created, so they don't follow what is now the standard LRO pattern. It also contrasts in an obvious way with StandardLro above :)

Reply:
That does sound good.